File: app/src/main/java/com/digis/android/task/data/NetInfoRepository.java
package com.digis.android.task.data;

import android.app.Application;

import com.digis.android.task.data.model.NetInfo;
import com.digis.android.task.data.service.MainService;
import com.digis.android.task.data.source.remote.RemoteDataSource;
import com.digis.android.task.data.source.remote.RemoteResponseHandler;
import com.digis.android.task.data.source.remote.Response;

import javax.inject.Inject;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

public final class NetInfoRepository extends BaseRepository {

    private final MainService mService;

    @Inject
    public NetInfoRepository(Application app, MainService service) {
        super(app);
        mService = service;
    }

    public LiveData<Response<NetInfo>> getNetInfo() {
        MutableLiveData<Response<NetInfo>> liveData = new MutableLiveData<>();
        RemoteDataSource<NetInfo> remoteSource = new RemoteDataSource<>(mService.getNetInfo());
        remoteSource.execute(new RemoteResponseHandler<>(getApp(), liveData));
        addRemoteDataSource(remoteSource);
        return liveData;
    }

}

File: app/src/main/java/com/digis/android/task/implementation/kotlin/data/NetInfoRepository.kt
package com.digis.android.task.implementation.kotlin.data

import androidx.lifecycle.LiveData
import com.digis.android.task.implementation.kotlin.data.model.NetInfo
import com.digis.android.task.implementation.kotlin.data.service.MainService
import com.digis.android.task.implementation.kotlin.data.source.remote.Response
import java.util.*
import javax.inject.Inject

class NetInfoRepository @Inject constructor(private val service: MainService) {
    fun getNetInfo(): LiveData<Response<NetInfo>> =
            service.getNetInfo(id = Random().nextInt(BOUND) + 1)
}

private const val BOUND = 30